packages/dds/tree/docs/wip/cell-model.md

```diff
@@ -0,0 +1,183 @@
+
+## The Model
+
+### Fields
+
+In this model, a field is a possibly empty sequence of cells.
+The sequence makes cells totally ordered.
+It is this ordering which serves as the foundation upon which the appropriate ordering of content is ensured.
+
+### Cells
+
+In this model, a cell is a unit of storage.
+A cell cannot be subdivided.
+It may either be empty or full.
+The contents of a (full) cell may be arbitrarily large or small (though a specific data model may constrain this in practice).
+
+> Note: the model does not prescribe for the cells to be explicitly reified.
+> While some cells may be represented at runtime some of the time, they are primarily a conceptual artefact.
+
+A cell may be annotated with forwarding information that specifies a target destination cell.
+A single cell may bear any number of such forwarding annotations.
+This forwarding information is used to represent move information: the source cell is annotated with forwarding information that points to the destination cell.
+
+In addition to their storage role, cells act as markers relative to which an edit may be performed (e.g., "Before/after cell foo").
+
+### Operations
+
+The model admits the following operations:
+
+> WIP: An alternative set of operations based on hiding cells instead of clearing them is being considered.
+> The general structure of the model is unaffected.
+
+- <u>Allocate</u> a new cell at a specific location in the sequence
+
+- <u>Fill</u> a cell with content (overwriting existing content if any)
+
+- <u>Clear</u> content from a cell, leaving it empty
+
+- <u>Add a forwarding annotation</u> to a cell
+
+- <u>Remove a forwarding annotation</u> from a cell
+
+Note that cells cannot be deallocated (or moved).
+This reflects the fact that a client may perform an edit relative to some content's location at a point in time, even if that content has since been deleted or moved: the client is performing the edit relative to the cell which contained that content.
+
+## Building on the Model
+
+The model can be used to describe low-level edits that a client may perform on a field in a collaborative environment and by so doing imply some of their merge semantics.
+
+We provide here two lists of such low-level edits, one for fields whose number of cells is fixed, and one for fields whose number of cells is dynamic.
+The dichotomy has no basis in the cell model but reflects what we think is a sensible separation and matches industry-standard patterns of editing.
+
+### Edits on Dynamically-Sized Fields
+
+Dynamically-sized fields can be thought of as having list-like behavior: elements can be added anywhere in the list and removed at will which makes the list dynamically grow and shrink.
+Dynamically-sized fields typically start out empty of any cell.
+
+The low-level edits that such fields might support can decomposed as follows:
+
+- Insert: allocate and fill a cell
+
+- Delete/Remove: clear a cell
+
+- Revive: fill a (cleared) cell with the contents it 
```

Comment:
Should this sentence be completed with "contained before"?

Reply:
Indeed! Thank you!